pkg/planner/cascades/memo/memo.go

```diff
@@ -230,3 +230,75 @@ func (mm *Memo) NewGroupExpression(lp base.LogicalPlan, inputs []*Group) *GroupE
 func (*Memo) mergeGroup(_, _ *Group) {
 	// todo: in next pull request
 }
+
+// IteratorLP serves as iterator to get all logical plan inside memo.
+type IteratorLP struct {
+	root      *Group
+	stackInfo []*list.Element
```

Comment:
The stack info is only used for debugging? Do we always need to operate it even if not in debugging mode?

Reply:
it's for iter position record, one element for each group, when return a LP back, next iteration can start with this position rather than start from 0.